### PR TITLE
documentation for list iterators

### PR DESCRIPTION
**Pull request by tobibeer**
Documentation for #1330...

* heavily modified **ListWidget**
    * extracted **NestedLists**
* added **IteratorVariables**

### CODE REVIEW

File: editions/tw5.com/tiddlers/demonstrations/NestedLists.tid [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+modified: 20150106180000000
+
+The following sidebar tabs give examples of grouped lists created by nesting.
+
+!! Types Tab
+
+The list in the [[Types|$:/core/ui/MoreSideBar/Types]] tab is generated with this markup:
+
+```
+<$list filter="[!is[system]has[type]each[type]sort[type]]">
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/demonstrations/NestedLists.tid]: Here perhaps we could transclude the actual content, something like:

```
<pre>
<code>
<$text text={{$:/core/ui/MoreSideBar/Types}}/>
</code>
</pre>
```

Comment (tobibeer) [inline editions/tw5.com/tiddlers/demonstrations/NestedLists.tid]: Shall we perhaps make a global wikitext helper macro of that for...

**$:/editions/tw5.com/wikitext-macros**

?

I'm not so fond of these macro names...

```
<<wikitext-example>>
<<wikitext-example-without-html>>
```

Perhaps make them...

```
<<example-html>>
<<example>>
```

?

And add...

```
<<example-code>>
```

Comment (Jermolene) [inline editions/tw5.com/tiddlers/demonstrations/NestedLists.tid]: I think that the macro name `<<example>>` is too generic, better leave that for users own macros. I'd call this macro `<<plain-text tiddler>>`. Also I just realised that we should use the codeblock widget. That way the text will be syntax highlighted if the plugin is installed:

```
\define plain-text(tiddler)
<$codeblock language={{$tiddler$!!type}}>
<$text text={{$tiddler$}}/>
</$codeblock>
```

Comment (tobibeer) [inline editions/tw5.com/tiddlers/demonstrations/NestedLists.tid]: Ok to add that to...

**$:/editions/tw5.com/wikitext-macros**

?

How about...

```
<<tw-demo>>
<<tw-demo-html>>
<<tw-code>>
<<tw-code-demo>>
```

?

Comment (Jermolene) [inline editions/tw5.com/tiddlers/demonstrations/NestedLists.tid]: @tobibeer if we really want to change the names of those existing macros it's better to discuss that in a fresh ticket, not these line notes (they're not as persistent as comments on an issue).

Comment (tobibeer) [inline editions/tw5.com/tiddlers/demonstrations/NestedLists.tid]: Ok, I'll just add the other macro for now.

**Comment by tobibeer** [comment]
Ok, added the macro(s) and updated the examples. For now, it doesn't implement the codeblock widget as you suggested here https://github.com/Jermolene/TiddlyWiki5/pull/1332#discussion_r22762157 because it would only return `undefined`.

**Comment by Jermolene** [comment]
> For now, it doesn't implement the codeblock widget as you suggested here #1332 (comment) because it would only return undefined.

I'd got the syntax wrong; it should be:

```
\define plain-text(tiddler)
<$codeblock language={{$tiddler$!!type}} code={{$tiddler$}}/>
```

**Comment by tobibeer** [comment]
Ok, thanks. Updated.

Can you perhaps write a little doc tiddler on the codeblock widget?
Thant would be nice.

**Comment by Jermolene** [comment]
Thanks @tobibeer 

> Can you perhaps write a little doc tiddler on the codeblock widget?

If you want to have a go at this while I'm away, have a look at this file:

https://github.com/Jermolene/TiddlyWiki5/blob/master/core/modules/widgets/codeblock.js

You can also see how the codeblock widget is overridden by the highlight plugin:

https://github.com/Jermolene/TiddlyWiki5/blob/master/plugins/tiddlywiki/highlight/highlightblock.js

**Comment by tobibeer** [comment]
Just so I understand how /prerelease works... how come **IteratorVariables** is not on there yet while other parts are? Just curious. I can imagine you're perhaps doing some rewording.

**Comment by Jermolene** [comment]
@tobibeer I've pushed a new prerelease with the file extension fixed in f4005871b4d1b84baec848b19afc2a5bdd3511d6

**Comment by tobibeer** [comment]
thx

**Pull request merged**